File: somijeon/2022-06-08-contains-duplicate.js
// Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.

//* Example 1:

// Input: nums = [1,2,3,1]
// Output: true

//* Example 2:

// Input: nums = [1,2,3,4]
// Output: false

const containsDuplicate = function (nums) {
  const mySet = new Set();
  for (let i = 0; i < nums.length; i++) {
    if (mySet.has(nums[i])) {
      return true;
    }
    mySet.add(nums[i]);
  }
  return false;
};

File: Euihyun Lee/week_1(0606-0610)/2022-06-08-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        i, nums_set = 0, set()
        while i < len(nums):
            j = len(nums_set)
            nums_set.add(nums[i])
            if j == len(nums_set):
                return True
            i += 1
        return False